dev/Installer/src/install.cpp

```diff
@@ -0,0 +1,172 @@
+#include "pch.h"
+#include "packages.h"
+#include "install.h"
+
+using namespace winrt;
+using namespace Windows::ApplicationModel;
+using namespace Windows::Foundation;
+using namespace Windows::Management::Deployment;
+using namespace Windows::System;
+
+winrt::hresult add_package(Uri packageUri)
+{
+    PackageManager packageManager;
+    auto deploymentOperation{ packageManager.AddPackageAsync(packageUri, nullptr, DeploymentOptions::None) };
+    deploymentOperation.get();
+    if (deploymentOperation.Status() != AsyncStatus::Completed)
+    {
+        return deploymentOperation.ErrorCode();
+    }
```

Comment:
Hey @DrusTheAxe, won't this `.get()` still will throw because a failed deployment operation emits a failed `HRESULT` from the `IAsyncOperationWithProgress<T,P>::GetResult`? So this unfortunately needs a try/catch, if that's the case.

Reply:
Wrapped both deployment operations in try-catch.